src/dbnode/storage/shard.go

```diff
@@ -842,14 +844,21 @@ func (s *dbShard) writeAndIndex(
 			return ts.Series{}, false, err
 		}
 	} else {
-		// This is an asynchronous insert and write
+		// This is an asynchronous insert and write which means we need to clone the annotation
+		// because it could get returned to a pool at any point after this function returns.
```

Comment:
nit: please rephrase `because it could get returned to a pool at any point after this function returns.` to something like `because its lifecycle in the commit log is independent of the calling function`